service/resource/endpoint/desired_test.go

```diff
@@ -1 +1,62 @@
 package endpoint
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/giantswarm/operatorkit/framework/context/canceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func Test_Resource_Endpoint_GetDesiredState(t *testing.T) {
+	testCases := []struct {
+		Obj              interface{}
+		ExpectedEndpoint interface{}
+	}{
+		{
+			Obj: &apiv1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "TestPod",
+					Namespace: "TestNamespace",
+					Annotations: map[string]string{
+						"endpoint.kvm.giantswarm.io/ip":      "1.1.1.1",
+						"endpoint.kvm.giantswarm.io/service": "TestService",
+					},
+				},
+			},
+			ExpectedEndpoint: []Endpoint{
+				{
+					IP:               "1.1.1.1",
+					ServiceName:      "TestService",
+					ServiceNamespace: "TestNamespace",
+				},
+			},
+		},
```

Comment:
I would like to see more test cases here as well.

Reply:
Done.